### DISCUSSION TITLE
Trigger body with foreach

### DISCUSSION THREAD

**Discussion by GuilhermeDeLucca** [LiveHelperChat/livehelperchat#1836] [General]
it is possible to use foreach, in the
Triggerbody?

I wanted/imagined doing something like this:

`[
    {
        "type": "text",
        "content": {
            "text": "Please select the ID number.",
            "quick_replies": [
             {foreach=content_1}                
               {,
                    "type": "trigger",
                    "content": {
                        "name": "{var_forache}",
                        "payload": "278",
                        "store_name": "var_storene_nem",
                        "store_value": "{var_forache}"
                    }
                },
             {/foreach}
            {}
            ]
            }
        }
    }
]`

I want to use foreach to assemble a list of buttons incremented by the API return natively by the LHC, without the need to do this treatment in the API

Is everything native through the LHC possible?

**Comment by remdex** [comment]
No, for buttons you can't do that, only for text message. So you will have to format in your end.

**Comment by GuilhermeDeLucca** [comment]
Thank you Remdex.

Please guide me with one more thing. Do I move this discussion, to Ideas (possible improvement for LHC native module), Q&A (since it's a simple answer) or leave it right here?

I will treat in the end as you said

**Reply by remdex** [reply]
Just leave it here. In general i think foreach should work, but because of comma you won't valid json after parse. You can also try to improve that part :)

**Reply by GuilhermeDeLucca** [reply]
You really can't implement {foreach} within the body of the trigger, so that it creates the buttons!

The only place {/foreach} is accepted would be within Text or Text HTML

Text HTML would have already resolved that it was my initial implementation because I can create the buttons, but I can't define Store Name to use as a variable in another trigger

**Comment by GuilhermeDeLucca** [comment]
I had thought about the comma issue so after closing the forche it passed {}

`},
{/foreach}
{}`

I was wondering if it would be some implementation with {{}} doubles or "{foreach}" but still without success I'll try some other idea and post again here.

**Reply by remdex** [reply]
In short I tested myself. You can't have foreach in that place, the only option is a new development (custom foreach just for trigger body trigger). Otherwise, this foreach would contradict with message body foreach and at the end we will have a mess...

**Comment by GuilhermeDeLucca** [comment]
That's what I thought!

So I leave you with a suggestion for future versions, to create an Enter JSON box in Send Text. So we can use foreach natively in the LHC or an implementation in Quick Reply to use the API return.

Enter HTML would even be possible to use if I managed to pass the Store Name (it is possible and I don't know)?

**Comment by remdex** [comment]
It has to be quick reply syntax not HTML in order for the `store value` and `store name` to work. As on button click I'm looking at message meta data to determine has it store value associated with it or not.

So I ques you will have to generate proper content in your Rest API. It's possible to develop this custom foreach which would apply only in trigger context, but would take 2-3 hours. If you are willing to develop it yourself correct place is here https://github.com/LiveHelperChat/livehelperchat/blob/master/lhc_web/lib/core/lhgenericbot/actionTypes/lhgenericbotactiontbody.php#L10

Or you can always sponsor development :)

**Comment by GuilhermeDeLucca** [comment]
Thank you for giving me the knife and cheese, now I'm going to cut it hahaha

remdex, thanks for your attention, I'll do it here and send it for your approval.

And you can be sure that soon I will sponsor the development of some improvement/functionality

Or worst case sponsor some beers and a supply of coffee LOL
